packages/extension/src/companion/CompanionPopupButton.tsx

```diff
@@ -29,6 +30,12 @@ export const CompanionPopupButton = (): ReactElement => {
     setShowCompanionPermission(!showCompanionPermission);
   };
 
+  const closeCompanionPopupButton = () => {
+    setShowCompanionPermission(false);
+  };
+
+  useKeyboardNavigation(window, [['Escape', closeCompanionPopupButton]]);
```

Comment:
While it is great to have this (also impressed you found a way how to utilize it), we might have to remove it for now since this will always have an event listener even when the popup is closed. Either we remove it as mentioned or update this hook to include another property on its 3rd parameter to indicate that no listener should be attached (an `enabled` boolean property for us to tell to return early).

Reply:
So would you prefer to remove it and find another way to achieve the close popup effect when pressing the 'esc' key? Or ignore this for now and focus only on close the popup when click outside?

Reply:
I would prefer if we focus now on fixing the outside click since that was the primary issue.

> So would you prefer to remove it and find another way to achieve the close popup effect when pressing the 'esc' key?

I actually prefer what you have found, we just need to refactor the hook to support what I explained above. If you feel like it, feel free to open another PR once we have this merged if you are up for the challenge.

Reply:
great, so I fixed it and removed what you asked for